fhir-model/src/main/java/com/ibm/fhir/model/util/ValidationSupport.java

```diff
@@ -298,6 +300,105 @@ public static void checkLanguageCodeableConcept(CodeableConcept language, String
+     */
+    public static void checkUcumCode(Code ucumCode, String elementName) {
+        if (ucumCode != null) {
+            String ucumCodeValue = ucumCode.getValue();
+            if (!UCUMUtil.isValidUcum(ucumCodeValue)) {
+               throw new IllegalStateException(String.format("'%s' is not a valid UCUM code", ucumCodeValue));
+            }
+        }
+    }
+
+    /**
+     * Checks that each UCUM coding in the list has a valid UCUM syntax.
+     * @param ucumCodings the UCUM coding list
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM coding list is not valid
+     */
+    public static void checkUcumCodings(List<Coding> ucumCodings, String elementName) {
+        if (ucumCodings != null) {
+            for (Coding ucumCoding : ucumCodings) {
+                checkUcumCoding(ucumCoding, elementName);
+            }
+        }
+    }
+    
+    /**
+     * Checks that the UCUM coding has a valid UCUM syntax.
+     * @param ucumCoding the UCUM coding
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM coding is not valid
+     */
+    public static void checkUcumCoding(Coding ucumCoding, String elementName) {
+        if (ucumCoding != null) {
+            if (hasSystemAndCodeValues(ucumCoding)) {
+                if (!UCUM_CODE_SYSTEM_URL.equals(ucumCoding.getSystem().getValue())) {
+                    throw new IllegalStateException(String.format("UCUM system is not '%s'", UCUM_CODE_SYSTEM_URL));
+                }
+                checkUcumCode(ucumCoding.getCode(), elementName);
+                return;
+            }
+            throw new IllegalStateException(String.format("'%s' does not contain a UCUM system of '%s' and a valid UCUM code", elementName, UCUM_CODE_SYSTEM_URL));
+        }
+    }
+
+    /**
+     * Checks that each UCUM codeable concept in the list has at least one coding with a valid UCUM syntax.
+     * @param ucumCodeableConcepts the UCUM codeable concept list
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM codeable concept list is not valid
+     */
+    public static void checkUcumCodeableConcepts(List<CodeableConcept> ucumCodeableConcepts, String elementName) {
+        if (ucumCodeableConcepts != null) {
+            for (CodeableConcept ucumCodeableConcept : ucumCodeableConcepts) {
+                checkUcumCodeableConcept(ucumCodeableConcept, elementName);
+            }
+        }
+    }
+
+    /**
+     * Checks that the UCUM codeable concept has at least one coding with a valid UCUM syntax.
+     * @param ucumCodeableConcept the UCUM codeable concept
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM codeable concept is not valid
+     */
+    public static void checkUcumCodeableConcept(CodeableConcept ucumCodeableConcept, String elementName) {
```

Comment:
It'd be great to turn this on / and off like we've done with some other validation features. 
That way we have an opportunity to still extract or tell someone how to extract bad data.

Reply:
We discussed doing this more widely - something we'd address in issue #1421 - and making sure it applies to the UCUM and language syntax validation as well.